### PR TITLE
KAFKA-15235: No test coverage reports for Java due to settings for Jacoco being incompatible with Gradle 8.x

### PR DESCRIPTION
**Pull request by bmscomp**
Fixing the Jacoco report generation issue, in the current Gradle version 8.x 

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -808,10 +809,9 @@ if (userEnableTestCoverage) {
     executionData.from = javaProjects.jacocoTestReport.executionData
 
     reports {
-      html.enabled = true
-      xml.enabled = true
+      xml.required = true
```
Comment (ex172000) [inline build.gradle]: Is there a reason to have two duplicated lines?

Comment (bmscomp) [reply build.gradle]: Tanks @ex172000  for the review, No reason to have a duplicate entry l will fix it now

Comment (bmscomp) [reply build.gradle]: @ex172000  I did not only duplicate the xml configuration I also removed the html configuration and this is fixed now

Comment (ex172000) [reply build.gradle]: Nice, thank you for the explanation! New to the project so asked to learn :)

**Review by ex172000 — APPROVED** [review]
👍

**Comment by divijvaidya** [comment]
Please add how did you test this and the before/after results. Thanks!

**Comment by bmscomp** [comment]
@divijvaidya  Before those changes, the generation of tests jacoco coverage report fails because of this error 

`"Could not set unknown property 'enabled' for Report html of type org.gradle.api.reporting.internal.TaskGeneratedSingleDirectoryReport"`

With the related changes, the generation of the coverage reports works and the reports (xml and html) are well generated in  `build/reports/jacoco/test/html/index.html` and `build/reports/jacoco/test/jacocoTestReport.xml ` paths

**Review by divijvaidya — APPROVED** [review]
I verified that running 
`./gradlew clients:reportCoverage -PenableTestCoverage=true -Dorg.gradle.parallel=false` on trunk fails with error `Could not set unknown property 'enabled' for Report html of type org.gradle.api.reporting.internal.TaskGeneratedSingleDirectoryReport`, hence, the motivation makes sense.

**Comment by divijvaidya** [comment]
Test failures are not related. Merging this one. Nice catch @bmscomp! I am surprised that this wasn't caught earlier. I will also backport this to 3.5.

**Pull request merged**